### ISSUE TITLE
a 'Fix' for royalroad i figured i would share (fixing misaligned/missing tables/colors)

### ISSUE THREAD

**Issue by Leifman35** [dteviot/WebToEpub#1076]
So i've been using web2epub on royalroad for quite a while, and while it works pretty damn well on most websites, i understand that royalroad is quite a different case with with a bunch of formatting/styling etc' that usually isn't happening elsewhere (other than the occasion bold/talics/etc' that aren't a challenge really) However with royalroad you have square tables/boxes with multiple different colors etc' and other stylish/effects that just don't translate at all when web2epub grabs those. at best you get a 'black borders' with white color boxes/tables and at worse you get completely miss-alligned and ones that just break the entire thing/look terrible.
The fix is rather simple, using royalroad's .css and what i've been doing is simply replacing the stylesheet.css in the .epub with a correct one that will make the end result look exactly like the webpage.
I figured i'll make this post and share it, and whoever it responsible for updating web2epub can check for themselves just how much of a difference it makes if you implement it in a next update and make sure it uses this .css and not whatever web2epub is grabbing now:

[stylesheet.zip](https://github.com/dteviot/WebToEpub/files/12640744/stylesheet.zip)

Figured it might be something i can 'give back' for using this awesome addon so much and loving it so much.

P.s. - awesome job on the 'skip author notes' or letting ppl enable them feature, it's super useful and something i always thought was a great addition, unlike never understanding why the 'style' sorta thingy never went over nor why web2epub didn't grab the proper .css for royalroad

**Comment by Leifman35** [comment]
and here are some further examples to show exactly what changes it does in regards to the stuff i mentioned, from 2 different stories:

1. Most 'standard' stories with 'blue boxes' would look like this by default on web2epub:
![Screenshot 2023-09-16 160227](https://github.com/dteviot/WebToEpub/assets/105802605/8fcdd563-3ace-48ad-903e-d9912139e4ec)

while if you use the .css i attached it would look like this:

![Screenshot 2023-09-16 160242](https://github.com/dteviot/WebToEpub/assets/105802605/9c889c49-bec8-459b-9a08-be5a31b6291f)

So not just the actual blue box/colors but also the 'centering' and getting the box dimensions right (usually web2epub just shows a white box with black borders but often messes up the dimensions too)

or if it is a story with more 'intricate' and specifically made colors/boxes, then they can totally just 'now show' right due to colors like in example 2:

![Screenshot 2023-09-16 160331](https://github.com/dteviot/WebToEpub/assets/105802605/6590a699-1d14-4417-9b93-22d12ad33dec)

This is defauly web2epub vs using the .css:

![Screenshot 2023-09-16 160337](https://github.com/dteviot/WebToEpub/assets/105802605/73f61a55-320a-45a8-bba6-72105f85adba)

or another one, to show the off-centring of the boxes, original web2epub:

![Screenshot 2023-09-16 160348](https://github.com/dteviot/WebToEpub/assets/105802605/23819676-aa6d-4ec6-ab47-f9a75ed745b1)

with the correct .css:

![Screenshot 2023-09-16 160407](https://github.com/dteviot/WebToEpub/assets/105802605/f52f2e20-a315-4a06-9134-38415ac42993)


Hope this helps! and i really hope it is implemented (not only for the fact i won't have to manually open every .epub file and replace the .css file, but overall i think it will make web2epub simply perfect for royalroad)

**Comment by Leifman35** [comment]
Or another idea is to make it similar to the "Remove Author Notes" button toggle, with something like "Keep original Boxes/Colors/View" , so one can select if to get the 'original vision' or the current (rather lacking) view that web2epub generates

**Comment by dteviot** [comment]
@Leifman35 

Thanks for the suggestion.  IIRC, the reason for the default black and white styling was that not all epub readers support color.
I think the correct solution would be have an option "include stylesheet from original source".

**Comment by gamebeaker** [comment]
@dteviot wouldn't this also solve the problem when text is scrambled via css instead of js?
Example: [https://secondlifetranslations.com/watf/watf-chapter-1/](https://secondlifetranslations.com/watf/watf-chapter-1/) (they have disabled right click on the page if you right click on paypal or ko-fi you can open inspector)
In the picture you can see a <span class="jmbl"> with scrambled text the scrambled font is defined in span.jmbl -> font-family: 'opne_sansscrambled'!important;
![grafik](https://github.com/dteviot/WebToEpub/assets/9269847/7f64934c-d5a3-4be6-8992-0fbd6c3b4ff2)

**Comment by Leifman35** [comment]
> @Leifman35
> 
> Thanks for the suggestion. IIRC, the reason for the default black and white styling was that not all epub readers support color. I think the correct solution would be have an option "include stylesheet from original source".

Yep. i completely agree on the 'option' as that was the latest i thought about it and even mentioned the "Remove Author Notes" as a similar sorta 'toggle' option as that i believe works amazingly and also isn't in the 'advanced settings', but rather in the main page when a person is generating the epub so it is easily spotted and can be toggled on or off.

I didn't know much about the other readers not supporting color, and i guess that makes sense why it was never implemented in the 'original way' per se, but still, it also fixes the formatting/correct centering so that is also a + :)
